### DISCUSSION TITLE
Autocomplete with vim.fn.input

### DISCUSSION THREAD

**Discussion by thisduck** [hrsh7th/nvim-cmp#1073] [Q&A]
Is there a way to get nvim-cmp to work with `vim.fn.input`?

I've tried to search for what the filetype is, but had no luck.

**Comment by alloc33** [comment]
If someone found any workaround - please share 😎

**Comment by sircaffe** [comment]
You can use ui input like so:
```lua
    vim.ui.input({
        prompt = 'input: ',
        completion = 'shellcmd',
    }, function(input)
        -- code here
    end)
```

completion options: https://neovim.io/doc/user/map.html#%3Acommand-completion

**Reply by cleerax** [reply]
The cmp plugin still doesn't work with ```vim.ui.input```. The default vim autocomplete is working at startup, but when cmp is loaded, autocomplete just stops working and instead it outputs ^I when I press tab while in input.

Upd: actually found a solution: https://github.com/hrsh7th/cmp-cmdline/issues/16#issuecomment-1299525844. Still doesn't complete using specified completion type though, just configured sources
